src/type.zig

```diff
@@ -1145,6 +1145,106 @@ pub const Type = extern union {
         }
     }
 
+    pub fn isGlobalErrorSet(ty: Type) bool {
```

Comment:
The logic for this is unfortunately not this simple. A Type could be a `error_set_inferred` in which case it *might* be the global error set. Determining whether an inferred error set is the global error set requires "resolving" the inferred error set, which could cause compile errors to be emitted. It means we cannot implement this function here.

Reply:
I have stubbed out isGlobalErrorSet and put it in Sema.zig for now, as it can return an error if the resolution fails.